internal/backend/runtime/omni/migration/migrations.go

```diff
@@ -1760,3 +1761,40 @@ func dropMachineClassStatusFinalizers(ctx context.Context, st state.State, logge
 
 	return nil
 }
+
+// createProviders creates infra.Provider resource for each existing infra.ProviderStatus.
+// This migration is required to avoid breaking user's providers connection which were created before 0.50.
+func createProviders(ctx context.Context, st state.State, logger *zap.Logger, _ migrationContext) error {
+	providers, err := safe.ReaderListAll[*infra.ProviderStatus](ctx, st)
```

Comment:
Use `auth.Identity` with prefix `access.InfraProviderPrefix` instead. 
This way it will be more resilient. Any providers ever registered will be able to connect after we roll out the change.

Reply:
Also test the providers connection manually.